### PR TITLE
Install certbot through pip

### PR DESCRIPTION
**Pull request by XeryusTC**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [x] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No. Although configuration has changed backwards compatibility has been preserved by adding some logic in `letsencrypt/map.jinja`.

### Describe the changes you're proposing
Install certbot by using pip. This is similar to using git except that it will download and install through pip. A virtualenv is created (default: `/opt/letsencrypt`).

### Pillar / config required to test the proposed changes
The `use_package` pillar variable is replaced by `install_method`. So for the system package installation method
```yaml
letsencrypt:
  use_package: true
```
becomes
```yaml
letsencrypt:
  install_method: package
```

Likewise for the git installation method. Additionally one can now do
```yaml
letsencrypt:
  install_method: pip
```
To use the new pip installation method.

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by myii** [comment]
Apologies for the delay, @XeryusTC.  I'm going to ask @javierbertoli to have a look at this, since there may be a breaking change in this.  The good thing is that all the CI instances are passing, including the new suite that you have provided.

File: letsencrypt/defaults.yaml [hunk 1]
```diff
@@ -2,25 +2,30 @@
 # vim: ft=yaml
 ---
 letsencrypt:
-  use_package: true
+  install_method: package
```
Comment (myii) [inline letsencrypt/defaults.yaml:5]: This change from `use_package` to `install_method` is the potential breaking change that I mentioned above.

Comment (litnialex) [reply letsencrypt/defaults.yaml:5]: Please check take a look at the letsencrypt/map.jinja regarding backwards compatibility..

**Review by javierbertoli — CHANGES_REQUESTED** [review]
@XeryusTC, It LGTM, nice addition.

I see there are a few conflicts now.

If you fix them, I think it's OK to merge.

File: letsencrypt/defaults.yaml (hunk 1 above)
Comment (javierbertoli) [reply letsencrypt/defaults.yaml:5]: As far as I can see, the PR is backward compatible, as @litnialex suggests.

**Review by javierbertoli — APPROVED** [review]

**Pull request merged**